### DISCUSSION TITLE
controlnet ++ support?

### DISCUSSION THREAD

**Discussion by hoodady** [Panchovix/stable-diffusion-webui-reForge#5] [General]
https://huggingface.co/xinsir/controlnet-union-sdxl-1.0/tree/main

**Comment by Panchovix** [comment]
Related to https://github.com/Panchovix/stable-diffusion-webui-reForge/issues/1, I want to add upstream support for controlnet, but it will be really hard by myself (I tried haha), but with help and more tries, I will try!

**Reply by hoodady** [reply]
Thank you for all you do, I realised it actually works, you just gotta go to all pick the controlnet unet first then canny or any pre processor!
